File: apps/agentic-workbench/src/entities/project/model/git-worktree.ts
// "unknown"은 status 계산을 건너뛴 경우(includeStatus: false) 또는 목록 도착 전
// placeholder 상태다(specs/007 research R4, R5).
export type GitWorktreeStatus = "clean" | "prunable" | "dirty" | "unknown";

export type GitWorktree = {
  path: string;
  head?: string | null;
  branch?: string | null;
  status: GitWorktreeStatus;
  pruneReason?: string | null;
  canDelete: boolean;
};

export type GitWorktreeCreateInput = {
  path: string;
  branch: string;
  reference: string;
};

/**
 * URL의 worktree 경로만으로 세션 화면을 즉시 렌더링하기 위한 placeholder.
 * 목록 응답이 도착하면 실제 메타데이터로 교체된다(specs/007 research R4).
 */
export function createPlaceholderWorktree(path: string): GitWorktree {
  return {
    path,
    head: null,
    branch: null,
    status: "unknown",
    pruneReason: null,
    canDelete: false,
  };
}

File: apps/agentic-workbench/src-tauri/src/domain/git_worktree.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GitWorktree {
    pub path: String,
    pub head: Option<String>,
    pub branch: Option<String>,
    pub status: GitWorktreeStatus,
    pub prune_reason: Option<String>,
    pub can_delete: bool,
}

#[derive(Debug, Clone, Serialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub enum GitWorktreeStatus {
    Clean,
    Prunable,
    Dirty,
    /// status 계산을 건너뛴 경우(`include_status=false`). clean/dirty 판정이
    /// 필요한 화면은 기본 옵션으로 조회한다(specs/007 research R5).
    Unknown,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GitWorktreeCreateDraft {
    pub path: String,
    pub branch: Option<String>,
    pub reference: Option<String>,
}
